chain/chain/src/chain.rs

```diff
@@ -1003,12 +1003,11 @@ impl Chain {
                         *block_hash,
                         GCMode::Canonical(tries.clone()),
                     )?;
-                    // TODO(resharding): Call clear_resharding_data once we figure out what's wrong
-                    // chain_store_update.clear_resharding_data(
-                    //     self.runtime_adapter.as_ref(),
-                    //     self.epoch_manager.as_ref(),
-                    //     *block_hash,
-                    // )?;
+                    chain_store_update.clear_resharding_data(
```

Comment:
Could we enable the resharding tests `test_shard_layout_upgrade_gc` and `test_shard_layout_upgrade_gc_v2`?